### PR TITLE
Tutorials Grouped By Section + Best Practices Broken Up

### PR DESCRIPTION
**Pull request by sabine**
This patch does two things:

1. Group tutorials by section ("Get Started", "Language", "Platform Tools", "Guides") so they can show up under the respective tab
2. Break up the "Best Practices" page into several "Platform Tools" tutorials.

As part of (1), we remove the HTMX navigation from the Learn area, since the sidebars do not scroll enough to warrant the complexity now.

This brings the Learn Homepage into a state where a layout according to Figma and the documentation rework can be implemented.

The tabs, for now, lead to the first tutorial in that section, except for the Platform/Exercises/Books pages, which have their own dedicated page.

|page|screenshot|
|-|-|
|Learn Homepage|Screenshot 2023-07-20 at 12-56-24 Learn OCaml|
|Get Started| Screenshot 2023-07-20 at 12-48-41 Get Up and Running With OCaml · OCaml Tutorials|
|Language|Screenshot 2023-07-20 at 12-48-45 Data Types and Matching · OCaml Tutorials|
|Platform Tools| Screenshot 2023-07-20 at 12-48-53 OCaml Platform|
|Guides|Screenshot 2023-07-20 at 12-48-49 Formatting and Wrapping Text · OCaml Tutorials|

### CODE REVIEW

File: data/tutorials/getting-started/gs_01_a_first_hour_with_ocaml.md [hunk 1]
```diff
@@ -4,8 +4,7 @@ title: Your First Day with OCaml
 description: >
   Discover the OCaml programming language in this longer tutorial that takes you
   from absolute beginner to someone who is able to write programs in OCaml.
-category: "getting-started"
-date: 2021-08-06T17:11:00-00:00
+category: "Getting Started"
```
Comment (tmattio) [inline data/tutorials/getting-started/gs_01_a_first_hour_with_ocaml.md:7]: I think it's better to keep IDs here and transform that to the actual text in the frontend.

**Review by tmattio — APPROVED** [review]
Thanks @sabine, that looks very good! I do have comments on the overall organisation, but since the documentation outline is still in progress, I don't think it's helpful to discuss this as part of this PR.
The only thing I feel semi-strongly about is the PPX guide under the Platform tool. Ppxlib is a platform tool, but PPX is a language concept. The guide would be better placed under language in my opinion.
This shouldn't block the merge, so I'm approving nonetheless and leaving it to you to decide what's best.

File: data/tutorials/getting-started/gs_01_a_first_hour_with_ocaml.md (hunk 1 above)
Comment (sabine) [reply data/tutorials/getting-started/gs_01_a_first_hour_with_ocaml.md:7]: Then we need to keep a map from ids to names somewhere in the frontend. I chose this way because it's more flexible and there's less places to update simultaneously when making changes. This is just the grouping of items in the sidebar.

Though, if it's important, I can change this.

**Comment by sabine** [comment]
I'll move the PPX guide to language.

File: data/tutorials/platform/bp_03_run_executables_and_tests.md [hunk 2]
```diff
@@ -1,7 +1,48 @@
...
+category: "Best Practices"
 ---
 
+# Running Executables and Tests with Dune
+
+## Running Executables
+
+> **TL;DR**
+> 
+> Add an `executable` stanza in your dune file and run the executable with `dune exec <executable_path>.exe` or `dune exec <public_name>`.
```
Comment (christinerose) [inline data/tutorials/platform/bp_03_run_executables_and_tests.md]:
```suggestion
> Add an `executable` stanza in your `dune` file and run the executable with `dune exec <executable_path>.exe` or `dune exec <public_name>`.
```

File: data/tutorials/platform/bp_03_run_executables_and_tests.md [hunk 3]
```diff
@@ -1,7 +1,48 @@
...
+
+To tell dune to produce an executable, you can use the executable stanza:
```
Comment (christinerose) [inline data/tutorials/platform/bp_03_run_executables_and_tests.md]:
```suggestion
To tell Dune to produce an executable, you can use the executable stanza:
```

File: data/tutorials/platform/bp_03_run_executables_and_tests.md [hunk 4]
```diff
@@ -1,7 +1,48 @@
...
+ (name <executable_name>)
+ (public_name <public_name>)
+ (libraries <libraries...>))
+```
+
+The `<executable_name>` is the name of the executable used internally in the project.
+The `<public_name>` is the name of the installed binary when installing the package.
+Finally, `<libraries...>` is the list of libraries to link to the executable.
+
+Once dune has produced the executable with `dune build`, you can execute it with `dune exec <executable_path>.exe` or `dune exec <public_name>`.
```
Comment (christinerose) [inline data/tutorials/platform/bp_03_run_executables_and_tests.md]:
```suggestion
Once Dune has produced the executable with `dune build`, you can execute it with `dune exec <executable_path>.exe` or `dune exec <public_name>`.
```

File: data/tutorials/platform/bp_03_run_executables_and_tests.md [hunk 5]
```diff
@@ -1,7 +1,48 @@
...
+
+For instance, if you've put your dune file in `bin/dune` with the following content:
```
Comment (christinerose) [inline data/tutorials/platform/bp_03_run_executables_and_tests.md]:
```suggestion
For instance, if you've put your `dune` file in `bin/dune` with the following content:
```

File: data/tutorials/platform/bp_03_run_executables_and_tests.md [hunk 6]
```diff
@@ -1,7 +1,48 @@
...
+```
+
+You can run it with `dune exec bin/main.exe` or `dune exec my-app`.
+
+## Running Tests
+
+
 > **TL;DR**
 > 
 > Add a `test` stanza in your dune file and run the tests with `dune build @runtest`.
```
Comment (christinerose) [inline data/tutorials/platform/bp_03_run_executables_and_tests.md]:
```suggestion
> Add a `test` stanza in your `dune` file and run the tests with `dune build @runtest`.
```

File: data/tutorials/platform/bp_04_create_libraries.md [hunk 7]
```diff
@@ -1,7 +1,13 @@
...
+description: |
+  How to create libraries with Dune
+category: "Best Practices"
 ---
 
+# Creating Libraries to Structure a Project with Dune
+
 > **TL;DR**
 > 
 > Add a `library` stanza in your dune file.
```
Comment (christinerose) [inline data/tutorials/platform/bp_04_create_libraries.md]:
```suggestion
> Add a `library` stanza in your `dune` file.
```

**Review by christinerose — APPROVED** [review]
